### PR TITLE
fix method POST

### PR DESCRIPTION
**Pull request by akalend**
fix a POST request in the evhtp_make_request() method

### CODE REVIEW

**Comment by adamel** [comment]
Should surround the check with:

**Comment by adamel** [comment]
Should surround the check with
    if (evbuffer_get_length(r->buffer_out)) {
        evbuffer_add_buffer(obuf, r->buffer_out);
    }
to be consistent with other functions.

**Comment by ellzey** [comment]
@adamel you have maintainer rights, when you think this is good to go; you can merge it.

File: evhtp.c [hunk 1]
```diff
@@ -243,7 +243,7 @@ status_code_to_str(evhtp_res code) {
 /**
  * @brief callback definitions for request processing from libhtparse
  */
-static htparse_hooks request_psets = {
+htparse_hooks request_psets = {
```
Comment (ellzey) [inline evhtp.c]: This can stay static, no reason to change it.

File: evhtp.h [hunk 2]
```diff
@@ -1352,6 +1352,13 @@ EVHTP_EXPORT int evhtp_make_request(evhtp_connection_t * c,
 
 EVHTP_EXPORT unsigned int evhtp_request_status(evhtp_request_t *);
 
+
+EVHTP_EXPORT void
```
Comment (ellzey) [inline evhtp.h]: we already have a hook for this: https://github.com/akalend/libevhtp/blob/develop/evhtp.c#L1920-L1922

File: evhtp.h [hunk 3]
```diff
@@ -1352,6 +1352,13 @@ EVHTP_EXPORT int evhtp_make_request(evhtp_connection_t * c,
...
+_evhtp_connection_eventcb(evbev_t * bev, short events, void * arg);
+
+EVHTP_EXPORT evhtp_connection_t *
+_evhtp_connection_new(evhtp_t * htp, evutil_socket_t sock, evhtp_type type);
```
Comment (ellzey) [inline evhtp.h]: exported functions shouldn't have a preceding underscore.

**Comment by ellzey** [comment]
also refer to what @adamel said.

**Comment by adamel** [comment]
Pushed the original commit with my suggested fix and an improved commit message as commit 8bc4ab38.

The additional commits added 8 days ago are obviously pure nonsense.

**Comment by ellzey** [comment]
Word, thanks

**Pull request closed without merging**